internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -312,9 +312,14 @@ func (a *Adapter) EnsureGlobalCandidateImageUpdated() (controller.OperationResul
 			return controller.ContinueProcessing()
 		}
 
-		// look for the expected snapshotComponnet and update
+		// update PromotedImage for all Components of GCL
+		// then look for the expected snapshotComponent and update
 		for _, snapshotComponent := range a.snapshot.Spec.Components {
 			snapshotComponent := snapshotComponent //G601
+			err = a.updateComponentLastPromotedImage(a.context, a.client, componentToUpdate, &snapshotComponent)
```

Comment:
Should this be executed under [this condition](https://github.com/konflux-ci/integration-service/pull/861/files#diff-cbf7e6d58cc12a67b910b5b9108ba68d206e75cc8560ca0ab2b19e2d34415702R323) only?
Meaning, we should only run this function when the `snapshotComponent`'s `name` is the same as the `componentToUpdate`.

Reply:
Yes I think you are correct. I was misinterpreting `Existing components on all clusters whose status.lastPromotedImage has not been updated`  but on further thought I believe you are correct and it should be only those relevant components.